src/main/java/seedu/address/logic/commands/ScheduleCommand.java

```diff
@@ -0,0 +1,93 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.candidate.Candidate;
+import seedu.address.model.interview.Interview;
+
+/**
+ * Schedules a candidate identified using it's displayed index from the address book for an interview
+ * on a specified time slot.
+ */
+public class ScheduleCommand extends Command {
+
+    public static final String COMMAND_WORD = "schedule";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Schedules the candidate identified by the index number for an interview on given date and time.\n"
+            + "Time and date given must not be in the past.\n"
+            + "Parameters: INDEX (must be a positive integer) + /at + DATE (in dd/mm/yyyy format)"
+            + "TIME (in hh:mm format)\n"
+            + "Example: " + COMMAND_WORD + " 1 /at 23-03-2022 13:30";
+
+    public static final String MESSAGE_INVALID_FORMAT_DATETIME =
+            "Date and/or Time is not in the following format: dd/MM/yyyy HH:mm";
+
+    public static final String MESSAGE_INVALID_DATETIME =
+            "Date and/or Time must not be in the past!";
+
+    public static final String MESSAGE_SCHEDULED_CANDIDATE_SUCCESS =
+            "Successfully scheduled %1$s %2$s for interview on %3$s %4$s";
+
+    public static final String MESSAGE_DUPLICATE_CANDIDATE_INTERVIEW =
+            "Interview for this candidate already exists!";
+
+    public static final String MESSAGE_CONFLICTING_INTERVIEW =
+            "Interview for another candidate clashes with the proposed time slot!";
+
+    private final Index targetIndex;
+    private final LocalDateTime interviewDateTime;
+
+    /**
+     * Creates a ScheduleCommand to schedule the candidate at specified index for an
+     * interview on {@code LocalDateTime}
+     */
+    public ScheduleCommand(Index targetIndex, LocalDateTime interviewDateTime) {
```

Comment:
Nice job on the `ScheduleCommand`! Well thought out! Team will definitely need to test it to sieve out anything to add.

Only comment: you might be missing the docs for `targetIndex` and `interviewDateTime`.